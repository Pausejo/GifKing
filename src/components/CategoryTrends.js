import React from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSync, faArrowTrendUp } from '@fortawesome/free-solid-svg-icons'
import { useTrends } from '../hooks/useTrends';

export const CategoryTrends = ({setCategories}) => {

    const {data: trends, loading} = useTrends();

    const trendClick = (trend) => {
        setCategories(cat => {
            return (!cat.find(c => c === trend.target.textContent)) ? 
                [trend.target.textContent, ...cat] : 
                cat
        });
    }

    return (<>
                <h3 className='trending'><FontAwesomeIcon icon={faArrowTrendUp} /> Trending</h3>
                
                <ol className="filters">
                    {
                        trends.map(t =>
                            <li key={t}>
                                <label onClick={trendClick}>{t}</label>
                            </li>)
                    }
                </ol>

                {loading && <FontAwesomeIcon icon={faSync} spin size='1x' />}
            </>
    )
}
