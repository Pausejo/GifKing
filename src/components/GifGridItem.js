import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({url, title}) => {

  const handleClick = () => {    
    navigator.clipboard.writeText(url);
  }

  return <div className="card" onClick={handleClick}>
            <h2 className="title">{title}</h2>
            <img src={url} alt={title} />
        </div>
}

GifGridItem.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
}