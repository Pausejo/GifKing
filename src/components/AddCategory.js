import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { CategoryTrends } from './CategoryTrends';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e)=>{
        setInputValue(e.target.value);        
    }

    const handleSubmit = (e)=>{
        e.preventDefault();

        if (inputValue.length>2) {
            setCategories(cat => {
                return (!cat.find(c => c === inputValue)) ? 
                    [inputValue, ...cat] : 
                    cat
            });
            setInputValue('');
        }
    }

    return (<>
            <form data-testid="form" onSubmit={handleSubmit}>
                <div className="row">
                    <input
                        className="balloon"
                        id="category"
                        type="text"
                        value={inputValue}
                        onChange={handleInputChange}
                        placeholder="Ingrese la categorÃ­a" />
                        <label htmlFor="category">Category</label>
                </div>
            </form>
            <CategoryTrends setCategories={setCategories}/>
        </>
    );
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
}