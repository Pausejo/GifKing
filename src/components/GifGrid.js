import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCircleMinus, faSync } from '@fortawesome/free-solid-svg-icons';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({category, setCategories}) => {
    const {data: images, loading} = useFetchGifs(category);

    const deleteCatagory = () => {        
        setCategories((c) => c.filter(i => i!== category));
    }

    return (
         <div role="GifGrid">
            <h2><label onClick={deleteCatagory}><FontAwesomeIcon className="delete" icon={faCircleMinus} /> {category}</label></h2>        

            {loading && <FontAwesomeIcon data-testid="loading" icon={faSync} spin size='4x' />}

            <div className="container">
                { images.map(img => <GifGridItem key={img.id} {...img} />) }            
            </div>
        </div>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
    setCategories: PropTypes.func.isRequired,
}