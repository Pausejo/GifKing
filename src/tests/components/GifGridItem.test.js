import React from 'react';
import renderer from 'react-test-renderer';
import { GifGridItem } from '../../components/GifGridItem';

describe('Testing GifGridItem', () => { 

    test('should show component correctly', () => { 
        const tree = renderer.create(<GifGridItem title='Hola' url='https://'/>).toJSON();

        expect(tree).toMatchSnapshot(); 
     })
 })