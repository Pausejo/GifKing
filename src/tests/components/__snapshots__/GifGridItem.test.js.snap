// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGridItem should show component correctly 1`] = `
<div
  className="card"
  onClick={[Function]}
>
  <h2
    className="title"
  >
    Hola
  </h2>
  <img
    alt="Hola"
    src="https://"
  />
</div>
`;
